website/docs/feature-flag-tutorials/use-cases/ai.md

```diff
@@ -0,0 +1,223 @@
+import { env } from "$env/dynamic/private";
+
+const openai = createOpenAI({
+    apiKey: env.OPENAI_API_KEY ?? "",
+});
+
+const variants = {
+    variant1: {
+        model: "gpt-3.5-turbo",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+    variant2: {
+        model: "gpt-4-turbo",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+    variant3: {
+        model: "gpt-4-vision-preview",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+};
+
+export const POST = async ({ request }) => {
+    const { messages } = await request.json();
+    const variant = variants["variant1"];
+    const result = await streamText({
+        model: openai(variant.model),
+        messages,
+        max_tokens: variant.max_tokens,
+        temperature: variant.temperature,
+    });
+    return result.toDataStreamResponse();
+};
+```
+
+This file is doing a few key things:
+
+1. Sets up our OpenAI client with an API key.
+2. Defines different AI model configurations in the `variants` object.
+3. Handles incoming chat requests with the `POST` function.
+
+When a request comes in, it:
+
+-   Extracts the messages from the request body.
+-   Selects a variant (currently hardcoded to 'variant1').
+-   Uses the OpenAI API to generate a response.
+-   Streams the response back to the client.
+
+The `streamText` function is one of the utilities provided by Vercel's AI SDK. It helps deal with the real-time streaming of the AI's responses.
+
+## Creating a feature flag with AI variants
+
+Instead of hardcoding `variant1`, we want to use feature flags to dynamically choose which AI model to use. This will let us easily switch between models, test different configurations, or even do some A/B testing to see which model performs better for which task.
+
+To implement this, we'll need to:
+
+1. Set up a feature flag provider (we'll use Unleash).
+2. Replace our static variant selection with feature flag calls.
+3. Use the feature flag in our code to determine which AI model and settings to use for each request.
+
+### Install a local feature flag provider
+
+In this section, we’ll install Unleash, run the instance locally, log in, and create a feature flag. If you prefer, you can use other tools instead of Unleash, but you’ll need to update the code accordingly.
+
+Use Git to clone the Unleash repository and Docker to build and run it. Open a terminal window and run the following commands:
+
+```sh
+git clone https://github.com/unleash/unleash.git
+cd unleash
+docker compose up -d
+```
+
+You now have Unleash installed on your machine and running in the background. You can access this instance in your web browser at [http://localhost:4242](http://localhost:4242).
+
+Log in to the platform using these credentials:
+
+```
+Username: admin
+Password: unleash4all
+```
+
+Click **New feature flag** to create a new feature flag.
+
+![Create a new feature flag](../ruby/new-ff.png)
+
+Create a new feature flag called `gpt-version`.
```

Comment:
It might be too much effort to change now, but I would have named the feature flag something like `ai-model` or something more generic, like `ai` or `ai-experiment`. The enabled status would control whether the assistant is active, and the variant would determine which model to use. This also adds an interesting vector to the guide: whether our experiment is enabled at all. For example, we might want to disable it if it stops working, we run out of credits, etc.